openff/evaluator/storage/localfile.py

```diff
@@ -14,19 +14,33 @@
 class LocalFileStorage(StorageBackend):
     """A storage backend which stores files in directories on the local
     disk.
+
+    Parameters
+    ----------
+    root_directory: str
+        The directory in which all stored objects are located.
+    cache_objects_in_memory: bool
+        If True, objects will be cached in memory after they are retrieved
+        from storage. This can be useful if you anticipate making queries
+        from the storage. However, it may use up a lot of memory.
     """
 
     @property
     def root_directory(self):
         """str: Returns the directory in which all stored objects are located."""
         return self._root_directory
 
-    def __init__(self, root_directory="stored_data"):
+    def __init__(
+        self, root_directory="stored_data", cache_objects_in_memory: bool = False
+    ):
         self._root_directory = root_directory
 
         if not path.isdir(root_directory) and len(root_directory) > 0:
             makedirs(root_directory)
 
+        self._cached_retrieved_objects = {}
```

Comment:
An annotation would be useful here for readability, especially while poking into only a small portion of the code. I think this might be like `dict[str, tuple[Any, str]]`?

Reply:
Sure, added a type annotation. Thanks!